### DISCUSSION TITLE
Angles between grain boundary direction and Y axis

### DISCUSSION THREAD

**Discussion by JinghaoXu92** [mtex-toolbox/mtex#1276] [Ask Anything] [answered]
Hi,

I want to visualize the angle between the grain boundary (segment) directions and one certain axis (north direction in specimen coordinate). While I was confused about how to define a certain axis.
What I had tried was:

```
ori = ebsd('id',gB_Ni.ebsdId).orientations;
axes = axis(ori(:,1),ori(:,2),'antipodal')
plot(gB_Ni,angle(gB_Ni.direction,axes)./degree,'linewidth',2)
```

I feel the definition of 'axes' is not the one I wanted. I want the 'axes' to be the north direction in specimen coordinate.

It will be very helpful if you have any inputs. I was using 5.3 edition.

Thanks!

**Comment by kilir** [comment, marked as answer]
Hi,
it looks like you are looking for something like that:
```
mtexdata titanium
grains = ebsd.calcGrains
grains = grains.smooth(8)
%%
gb = grains.boundary('indexed')
plot(gb, angle(gb.direction, vector3d.X)/degree)
mtexColorMap hsv
```
Cheers,
Rüdiger

**Reply by JinghaoXu92** [reply]
Hi Rüdiger, I appreciate the input.
It works very well with 
`(gb.direction, vector3d.X)/degree)`

Best,
Jinghao
